jsapp/js/account/organization/MemberRemoveModal.tsx

```diff
@@ -0,0 +1,118 @@
+// Partial components
+import Button from 'jsapp/js/components/common/button';
+import InlineMessage from 'jsapp/js/components/common/inlineMessage';
+import KoboModal from 'jsapp/js/components/modals/koboModal';
+import KoboModalHeader from 'jsapp/js/components/modals/koboModalHeader';
+import KoboModalContent from 'jsapp/js/components/modals/koboModalContent';
+import KoboModalFooter from 'jsapp/js/components/modals/koboModalFooter';
+
+// Stores, hooks and utilities
+import {getSimpleMMOLabel} from './organization.utils';
+import envStore from 'jsapp/js/envStore';
+import subscriptionStore from 'jsapp/js/account/subscriptionStore';
+import {useRemoveOrganizationMember} from './membersQuery';
+import {notify} from 'alertifyjs';
+
+export const REMOVE_SELF_TEXT = {
+  title: t('Leave this ##team/org##'),
+  description: t('Are you sure you want to leave this ##team/org##?'),
+  dangerMessage: t('You will immediately lose access to any projects owned by this ##team/org##. This action cannot be undone.'),
+  confirmButtonLabel: t('Leave ##team/org##'),
+};
+
+export const REMOVE_MEMBER_TEXT = {
+  title: t('Remove ##username## from this ##team/org##'),
+  description: t('Are you sure you want to remove ##username## from this ##team/org##?'),
+  dangerMessage: t('Removing them from this ##team/org## also means they will immediately lose access to any projects owned by your ##team/org##. This action cannot be undone.'),
+  confirmButtonLabel: t('Remove member'),
+};
+
+/**
+ * Replaces placeholders with values.
+ * Note: assumes all placeholders want to be lowercase.
+ */
+function replacePlaceholders(text: string, username: string, mmoLabel: string) {
+  return text
+    .replaceAll('##username##', username)
+    .replaceAll('##team/org##', mmoLabel);
+}
+
+interface MemberRemoveModalProps {
+  username: string;
+  isRemovingSelf: boolean;
+  onConfirmDone: () => void;
+  onCancel: () => void;
+}
+
+/**
+ * A confirmation prompt modal for removing a user from organization. Displays
+ * two buttons and warning message.
+ *
+ * Note: it's always open - if you need to hide it, just don't render it at
+ * the parent level.
+ */
+export default function MemberRemoveModal(
+  {
+    username,
+    isRemovingSelf,
+    onConfirmDone,
+    onCancel,
+  }: MemberRemoveModalProps
+) {
+  const removeMember = useRemoveOrganizationMember();
+  const mmoLabel = getSimpleMMOLabel(
+    envStore.data,
+    subscriptionStore.activeSubscriptions[0],
+    false,
+    false
+  );
+
+  // Choose proper text
+  let title = isRemovingSelf ? REMOVE_SELF_TEXT.title : REMOVE_MEMBER_TEXT.title;
```

Comment:
1. I think we could merge all this into something like `let textToDisplay = isRemovingSelf ? REMOVE_SELF_TEXT : REMOVE_MEMBER_TEXT`. 

2. Below, instead of calling replacePlaceholders() four times individually you could iterate over the object.

Reply:
I simplified this following your both of your advices